core/src/test/java/org/apache/calcite/util/SourceTest.java

```diff
@@ -67,6 +72,32 @@ private static String getRootPrefix() {
     }
   }
 
+  @Test void testRelativeFileToUrl() throws UnsupportedEncodingException {
+    // a file path with space
+    final String path = "abc def.txt";
+    Source fileSource = file(null, path);
+    URL url = fileSource.url();
+
+    assertNotNull(url, "No url generated for file source");
+
+    String expected = "file:" + URLEncoder.encode(path, StandardCharsets.UTF_8.name());
```

Comment:
Would you please use literal for the expected value? It would make the expected value easier to understand when reading the code.

Reply:
Sure. Thank you for the good suggestion.